poly_commit/src/hyrax/pedersen.rs

```diff
@@ -0,0 +1,65 @@
+use arith::FieldSerde;
+use halo2curves::{
+    ff::{Field, PrimeField},
+    group::Curve,
+    msm, CurveAffine,
+};
+
+use crate::StructuredReferenceString;
+
+#[derive(Clone, Debug, Default)]
+pub struct PedersenParams<C: CurveAffine + FieldSerde> {
+    pub bases: Vec<C>,
+}
+
+impl<C: CurveAffine + FieldSerde> FieldSerde for PedersenParams<C> {
+    const SERIALIZED_SIZE: usize = unimplemented!();
+
+    fn serialize_into<W: std::io::Write>(&self, mut writer: W) -> arith::FieldSerdeResult<()> {
+        self.bases.serialize_into(&mut writer)
+    }
+
+    fn deserialize_from<R: std::io::Read>(mut reader: R) -> arith::FieldSerdeResult<Self> {
+        let bases: Vec<C> = Vec::deserialize_from(&mut reader)?;
+        Ok(Self { bases })
+    }
+}
+
+impl<C: CurveAffine + FieldSerde> StructuredReferenceString for PedersenParams<C> {
+    type PKey = Self;
+    type VKey = Self;
+
+    fn into_keys(self) -> (Self::PKey, Self::VKey) {
+        (self.clone(), self)
+    }
+}
+
+pub(crate) fn pedersen_setup<C: CurveAffine + FieldSerde>(
+    length: usize,
+    mut rng: impl rand::RngCore,
+) -> PedersenParams<C>
+where
+    C::Scalar: PrimeField,
+{
+    let bases: Vec<C> = (0..length)
+        .map(|_| {
+            let scalar = C::Scalar::random(&mut rng);
+            (C::generator() * scalar).to_affine()
```

Comment:
collect all the projective elements first and then use `batch_affine`

Reply:
sure - see https://github.com/PolyhedraZK/Expander/pull/200/commits/48b0df2dafa16dfac526d28057ad0b0827ca8444